sonic-net/sonic-mgmt
# [arp]: Extend proxy_arp to cover dualtor scenario

Signed-off-by: Lawrence Lee <lawlee@microsoft.com>


### Description of PR


Summary:
Fixes # (issue)

### Type of change





### Back port request

### Approach
#### What is the motivation for this PR?
Missing coverage for dual ToR scenario where two different servers under the same ToR pair have different active ToRs and are trying to communicate with each other. Prior to the image fix (https://github.com/sonic-net/sonic-swss/pull/2334), if the route for a neighbor was pointing to the IPinIP tunnel, the ToR would not proxy ARP for that neighbor.

#### How did you do it?
- Rename `test_arp_dualtor.py` to `test_arp_extended.py`
- Create new file `test_arp_dualtor.py` specifically for dual ToR scenarios
    - Add test to verify proxy ARP behavior for target IPs learned on standby links
    - Skip this test on non 202012 devices since it depends on `accept_untracked_na` which is currently only available in 202012
- Extend GARP service to send NA messages
- Edit PR tests/kvm tests to use the new `test_arp_extended.py` name

#### How did you verify/test it?
- Run `test_arp_dualtor.py` test case on dualtor running 202012, verify it passes
- Run `test_arp_dualtor.py` test case on dualtor on non-202012, verify it is skipped
- Run `test_arp_extended.py` on single ToR and dualtor, verify both pass.
- Turn off the fix for this issue, verify the `test_arp_dualtor.py` fails

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
